### PR TITLE
JSON encoding of Java codegen classes

### LINKED ISSUE #15868: Utility / library for json serilization of output type of java codegen
From https://github.com/digital-asset/daml/issues/13558

Potentially we will need these dependencies to implement the json serilization.

 - `//ledger/ledger-api-common` (`NoLoggingValueValidator.validateValue`)
 - `ledger-api-java` (`javaapi.data.Value`)
 - `lf-value-json` (`ApiCodecCompressed`)
 - `daml-lf/interface` (signature types)

### PR DESCRIPTION
**Pull request by chunlokling-da**
fixes #15868

### CODE REVIEW

**Review by ray-roestenburg-da — APPROVED** [review]

File: language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.daml.lf.value.json.ApiCodecCompressed
+import com.daml.lf.value.json.ApiCodecCompressed.JsonImplicits._
+import spray.json.JsValue
+
+final class JsonCodec(encodeDecimalAsString: Boolean, encodeInt64AsString: Boolean) {
+  val apiCodec = new ApiCodecCompressed(encodeDecimalAsString, encodeInt64AsString)
+
+  /** Converts `javaapi.data.Value` to `JsValue`. */
+  def toJsValue(value: Value): JsValue =
+    apiCodec.apiValueToJsValue(ValueConversion.toLfValue(value))
```
Comment (S11001001) [inline language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala:16]: I suggest another function that takes `DefinedDataType[_]` as an argument as well.

File: language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+object JsonCodec {
+  def apply() = new JsonCodec(false, false)
```
Comment (S11001001) [inline language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala:20]:
```suggestion
  val encodeAsNumbers = new JsonCodec(false, false)
```

File: language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.codegen.json
+
+import com.daml.ledger.javaapi.data.Value
+import com.daml.lf.value.json.ApiCodecCompressed
+import com.daml.lf.value.json.ApiCodecCompressed.JsonImplicits._
+import spray.json.JsValue
+
+final class JsonCodec(encodeDecimalAsString: Boolean, encodeInt64AsString: Boolean) {
```
Comment (S11001001) [inline language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala:11]:
```suggestion
final class JsonCodec private(encodeDecimalAsString: Boolean, encodeInt64AsString: Boolean) {
```

File: language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  val apiCodec = new ApiCodecCompressed(encodeDecimalAsString, encodeInt64AsString)
```
Comment (S11001001) [inline language-support/java/json/src/main/scala/com/daml/lf/codegen/json/JsonCodec.scala:12]:
```suggestion
  private val apiCodec = new ApiCodecCompressed(encodeDecimalAsString, encodeInt64AsString)
```

**Review by S11001001 — APPROVED** [review]

**Pull request merged**